bindsnet/environment/__init__.py

```diff
@@ -121,27 +121,35 @@ def step(self, a: int) -> Tuple[torch.Tensor, float, bool, Dict[Any, Any]]:
 
         self.preprocess()
 
+        # Add the raw observation from the gym environment into the info
+        # for debugging and display
+        info['gym_obs'] = self.obs
+
         # Store frame of history and encode the inputs.
         if len(self.history) > 0:
             self.update_history()
             self.update_index()
-
-        raw_obs = self.obs
+            # add the delta observation into the info for debugging and
+            # display
+            info['delta_obs'] = self.obs
 
         # The new standard is BxTxCxHxW. The gym environment doesn't
         # follow exactly the same protocol.
-        self.obs = self.encoder(self.obs).unsqueeze(1)
-        if self.obs.dim() == 4:
-            self.obs = self.obs.unsqueeze(2)
+
+        # we want CxHxW. If it is in HxW add in  the channel dimension
```

Comment:
What if the data is 1D?

Reply:
Updated the code to include comments on exact assumptions as well as adding a flag in the constructor to not automatically add the channel dimension for 2D observations